src/App.tsx

```diff
@@ -1,61 +1,41 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { NewTodo } from './components/NewTodo';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoUser } from './types/TodoUser';
+
+function assignData(users: User[], todos: Todo[]): TodoUser[] {
+  const newData: TodoUser[] = [];
+
+  users.forEach(user => {
+    todos.forEach(todo => {
+      if (user.id === todo.userId) {
+        newData.push(Object.assign({ ...todo }, { user: user }));
+      }
+    });
+  });
+
+  return newData;
+}
+
+const list = assignData(usersFromServer, todosFromServer);
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(list);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <NewTodo users={usersFromServer} todos={todoList} onAdd={setTodoList} />
```

Comment:
Prepare data in one place (App component) and pass it to child components. Don't use `import` keyword in non-root components. You are passing `usersFromServer` directly to the `NewTodo` component. It would be better to prepare all the data in the `App` component and pass it down to the child components.